src/main/java/com/epam/controller/ToursController.java

```diff
@@ -12,22 +15,35 @@
 import org.springframework.web.servlet.ModelAndView;
 
 import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 
 @Controller
 @Slf4j
 public class ToursController {
     private final TourOfferService toursOfferService;
+    private final HotelService hotelService;
+    private final ReservationService reservationService;
 
     @Autowired
-    ToursController(TourOfferService toursOfferService) {
+    ToursController(TourOfferService toursOfferService, HotelService hotelService, ReservationService reservationService) {
         this.toursOfferService = toursOfferService;
+        this.hotelService = hotelService;
+        this.reservationService = reservationService;
     }
 
     @GetMapping("/listoftours")
     public ModelAndView getToursList() {
         ModelAndView toursModel = new ModelAndView();
-        toursModel.addObject("list", toursOfferService.getTours());
+        List<Hotel> listOfHotels = hotelService.getHotels();
```

Comment:
It is really hard to understand this logic, can you write unit test to explain, what is happen here?
Why we need these changes?

Reply:
Okey, I’ll do it.
I used method from service of Hotel to pull list of hotels, which needed to show full information about tour.

Reply:
add unit test for it please, and this logic can be in service layer